src/bson.c

```diff
@@ -767,14 +796,66 @@ static const bson_visitor_t php_bson_visitors = {
 	{ NULL }
 };
 
+static bool map_element_matches_field_path(php_phongo_field_path_map_element* map_element, php_phongo_field_path* current)
+{
+	size_t i;
+
+	if (map_element->entry->size != current->size) {
+		return false;
+	}
+	for (i = 0; i < current->size; i++) {
+		if (strcmp(map_element->entry->elements[i], "$") == 0) {
+			continue;
```

Comment:
I imagine this is responsible for my earlier questions about wildcards have less precedence than an exact field name, despite appearing earlier in the list of field paths.

I do think it's more intuitive to follow first-match-wins logic here. If we match a wildcard or an exact field in this `for` iteration, return true immediately. If nothing matches, we can return `false` after iteration has succeeded.

I realize that the current implementation can return early. If you're concerned with overhead, we can always benchmark and optimize down the line.

Reply:
It currently implements the "first-match-wins" logic. If we find a wild card, we instantly skip to the next *level*, and not to the next element in the field path that we're trying to match against.

Reply:
I did actually run some benchmarks, as you can see in ``typemap-008.phpt``. The ``25000`` is there for a reason.